File: internal/content/testdata/bench/javascript/nc-daily-temperatures.js
function daily_temperatures(temps) {
    const result = new Array(temps.length).fill(0);
    const stack = []; // [temperature, index]
    for (let i = 0; i < temps.length; i++) {
        const t = temps[i];
        while (stack.length > 0 && t > stack[stack.length - 1][0]) {
            const [, j] = stack.pop();
            result[j] = i - j;
        }
        stack.push([t, i]);
    }
    return result;
}

File: internal/content/testdata/bench/rust/nc-daily-temperatures.rs
fn daily_temperatures(temps: Vec<i64>) -> Vec<i64> {
    let mut result = vec![0i64; temps.len()];
    let mut stack: Vec<(i64, usize)> = Vec::new(); // (temperature, index)
    for (i, &t) in temps.iter().enumerate() {
        while let Some(&(top_t, j)) = stack.last() {
            if t > top_t {
                stack.pop();
                result[j] = (i - j) as i64;
            } else {
                break;
            }
        }
        stack.push((t, i));
    }
    result
}
